src/NServiceBus.Core/Unicast/UnicastBusInternal.cs

```diff
@@ -27,20 +27,11 @@ partial class UnicastBusInternal : IStartableBus
         /// <summary>
         /// Initializes a new instance of <see cref="UnicastBus"/>.
         /// </summary>
-        public UnicastBusInternal(
-            BehaviorContextStacker contextStacker,
-            IMessageMapper messageMapper,
-            IBuilder builder,
-            ReadOnlySettings settings)
+        public UnicastBusInternal(IMessageMapper messageMapper, IBuilder builder, ReadOnlySettings settings)
         {
             this.settings = settings;
             this.builder = builder;
-
-            busImpl = new ContextualBus(
-                contextStacker,
-                messageMapper,
-                builder,
-                settings);
+            busImpl = new ContextualBus(new BehaviorContextStacker(builder), messageMapper, builder, settings);
```

Comment:
The usage of `ContextualBus` in `TransportReceiver` requires us to create the instance of `BehaviorContextStacker` outside (since we need to register it in the container).